webapp/src/compiler.ts

```diff
@@ -550,11 +564,15 @@ async function getCachedApiInfoAsync(project: pkg.EditorPackage, bundled: pxt.Ma
         jres: {}
     };
 
-    for (const used of usedInfo) {
+    for (const used of usedPackageInfo) {
+        const { info, dirname } = used;
         if (!used) continue;
-        pxt.Util.jsonCopyFrom(result.byQName, used.apis.byQName);
+        pxt.Util.jsonCopyFrom(result.byQName, info.apis.byQName);
+        for (const api of Object.keys(info.apis.byQName)) {
+            result.byQName[api].pkg = dirname;
```

Comment:
Isn't the loop overwriting this in each iteration?

Reply:
i.e. for each used package

Reply:
it's replacing the pkg of each one that has just been copied from info.apis.byQName into result; it's probably confusing to iterate over one map (the package) and update another (the output api info), I'll switch the order and update info.apis.byQName before copying